src/modules/accounting/accounting.py

```diff
@@ -1,31 +1,169 @@
+        validators_count, cl_balance = self._get_consensus_lido_state(blockstamp)
+
+        finalization_share_rate = self._get_finalization_shares_rate(blockstamp)
+        last_withdrawal_id_to_finalize = self._get_last_withdrawal_request_to_finalize(blockstamp)
+
+        exited_validators = self.lido_validator_state_service.get_lido_new_exited_validators(blockstamp)
+
+        # Here report all exited validators even they were reported before.
+        if exited_validators:
+            stacking_module_id_list, exit_validators_count = zip(*exited_validators.items())
+        else:
+            stacking_module_id_list = exit_validators_count = []
+
+        extra_data = self.lido_validator_state_service.get_extra_data(blockstamp)
+
+        # Filter stucked and exited validators that was previously reported
+        report_data = ReportData(
+            consensus_version=self.CONSENSUS_VERSION,
+            ref_slot=blockstamp.ref_slot,
+            validators_count=validators_count,
+            cl_balance_gwei=cl_balance,
+            stacking_module_id_with_exited_validators=stacking_module_id_list,
+            count_exited_validators_by_stacking_module=exit_validators_count,
+            withdrawal_vault_balance=self._get_withdrawal_balance(blockstamp),
+            el_rewards_vault_balance=self._get_el_vault_balance(blockstamp),
+            last_withdrawal_request_to_finalize=last_withdrawal_id_to_finalize,
+            finalization_share_rate=finalization_share_rate,
+            is_bunker=self._is_bunker(blockstamp),
+            extra_data_format=extra_data.format,
+            extra_data_hash=extra_data.data_hash,
+            extra_data_items_count=extra_data.items_count,
+        )
+
+        return report_data
+
+    @lru_cache(maxsize=1)
+    def _get_consensus_lido_state(self, blockstamp: BlockStamp) -> tuple[int, Gwei]:
+        lido_validators = self.w3.lido_validators.get_lido_validators(blockstamp)
+
+        count = len(lido_validators)
+        balance = Gwei(sum(int(validator.validator.balance) for validator in lido_validators))
+        return count, balance
+
+    def _get_withdrawal_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.withdrawalVault().call(),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_el_vault_balance(self, blockstamp: BlockStamp) -> Wei:
+        return Wei(self.w3.eth.get_balance(
+            self.w3.lido_contracts.lido_locator.functions.elRewardsVault().call(),
+            block_identifier=blockstamp.block_hash,
+        ))
+
+    def _get_last_withdrawal_request_to_finalize(self, blockstamp: BlockStamp) -> int:
+        return 0
+
+    def _get_finalization_shares_rate(self, blockstamp: BlockStamp) -> int:
+        # handleOracleReport
+        last_ref_slot = self.report_contract.functions.getLastProcessingRefSlot().call(
```

Comment:
need to handle a case if there was no report yet. i guess we can safely not to finalize any requests

Reply:
Fixed